client/shipping-label/views/purchase/steps/rates/test/list.js

```diff
@@ -0,0 +1,384 @@
+			} );
+			it( 'should display error', () => {
+				const errorWrapper = wrapper.find( FieldError );
+				// const errorText = errorWrapper.text();
+				expect( errorWrapper ).to.have.length( 1 );
+				expect( errorWrapper.prop( 'text' ) ).to.equal( 'There was an error!' );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+
+		describe( 'Both rates and errors', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...singlePackageProps }
+					availableRates={ {
+						weight_0_individual: Object.assign( {}, package_1_rate, package_1_error ),
+					} }
+					selectedRates={ {
+						weight_0_individual: '',
+					} }
+					errors={ {
+						server: {
+							weight_0_individual: [ {
+								message: 'There was an error!',
+							} ],
+						},
+						form: {
+							weight_0_individual: null,
+						},
+					} }
+				/>
+			);
+			it( 'should display rates drop down', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 1 );
+			} );
+			it( 'should display error', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 1 );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+	} );
+
+	describe( 'Multiple packages', () => {
+		describe( 'Rates only for all packages', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...multiPackageProps }
+					availableRates={ {
+						weight_0_individual: package_1_rate,
+						weight_1_individual: package_2_rate,
+					} }
+					errors={ {} }
+				/>
+			);
+			it( 'should display dropdown for each package', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 2 );
+			} );
+			it( 'should not display any errors', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 0 );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+
+		describe( 'Errors only for all packages', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...multiPackageProps }
+					availableRates={ {
+						weight_0_individual: package_1_error,
+						weight_1_individual: package_2_error,
+					} }
+				/>
+			);
+			it( 'should not display any dropdowns', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 0 );
+			} );
+			it( 'should display error for each package', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 2 );
+			} );
+			it( 'should display extra heading for package title', () => {
+				const packageNames = wrapper.find( 'p' ).map( node => node.text() );
+				expect( wrapper.find( 'p' ) ).to.have.length( 2 );
```

Comment:
Can another `.find()` call be avoided here by reusing the `packageNames` variable? Or perhaps can this line be omitted in favor of the line below?